crates/telio-wg/src/link_detection/pinger.rs

```diff
@@ -0,0 +1,121 @@
+use async_trait::async_trait;
+use socket2::Type;
+use std::net::{Ipv4Addr, Ipv6Addr};
+use std::sync::Arc;
+use std::time::Duration;
+use std::{convert::TryInto, net::IpAddr};
+use surge_ping::{
+    Client, Config as PingerConfig, ConfigBuilder, PingIdentifier, PingSequence, ICMP,
+};
+
+use telio_crypto::PublicKey;
+use telio_task::{
+    io::{chan, mc_chan, Chan, McChan},
+    task_exec, ExecError, Runtime, RuntimeExt, Task, WaitResponse,
+};
+use telio_utils::{telio_log_debug, telio_log_error, DualTarget};
+
+/// Component used to check the link state when we think it's down.
+/// We can check it before we announce it as down.
+///
+/// Can be used with both IPv4 and IPv6 addresses.
+pub struct Pinger {
+    task: Task<State>,
+}
+
+impl Pinger {
```

Comment:
This file is a bit similar to `crates/telio-nurse/src/rtt/ping.rs`. Maybe it would be wise to export it to some helper object with API, which satisfies both use cases. But no hard opinion here though ....

Reply:
Doubling down on this comment - the code is very duplicate of another file and becasue it contains more stuff like some macos/ios/tvos conditionals I highly suggest reusing the existing pinger instead